grammars/css.cson

```diff
@@ -20,294 +20,1205 @@
+          }
+        ]
+      }
+    ]
+  'functions':
+    'patterns': [
+      # Calculation
+      {
+        'begin': '(?i)(?<![\\w-])(calc)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.calc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.calc.css'
+        'patterns': [
+          {
+            'match': '[*/]|(?<=\\s|^)[-+](?=\\s|$)'
+            'name': 'keyword.operator.arithmetic.css'
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Colours
       {
-        'include': '#string-single'
+        'begin': '(?i)(?<![\\w-])(rgba?|hsla?)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.misc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.color.${1:/downcase}-value.css'
+        'patterns': [
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Gradients
       {
-        'include': '#comment-block'
+        'begin': '''(?xi) (?<![\\w-])
+          (
+            (                            # Captured for scope-name
+              (?:-webkit-|-moz-|-o-)?    # Accept prefixed/historical variants
+              (?:repeating-)?            # "Repeating"-type gradient
+              (?:linear|radial|conic)    # Shape
+            )
+            -gradient
+          )
+          (\\()
+        '''
+        'beginCaptures':
+          '1':
+            'name': 'support.function.gradient.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.${2:/downcase}.gradient.css'
+        'patterns': [
+          {
+            'match': '(?i)(?<![\\w-])(from|to|at)(?![\\w-])'
+            'name': 'keyword.operator.gradient.css'
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
+      }
+      # Gradients (Deprecated/original Webkit syntax)
+      {
+        'begin': '(?i)(?<![\\w-])(-webkit-gradient)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'invalid.deprecated.gradient.function.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.legacy-webkit.gradient.css'
```

Comment:
I would include `.deprecated` somewhere in this name, maybe `meta.function.gradient.invalid.deprecated.css`.